cli/tests/integration/npm_tests.rs

```diff
@@ -1054,6 +1054,181 @@ fn auto_discover_lock_file() {
   ));
 }
 
+#[test]
+fn peer_deps_with_copied_folders_and_lockfile() {
+  let _server = http_server();
+
+  let deno_dir = util::new_deno_dir();
+  let temp_dir = util::TempDir::new();
+
+  // write empty config file
+  temp_dir.write("deno.json", "{}");
+  let test_folder_path = test_util::testdata_path()
+    .join("npm")
+    .join("peer_deps_with_copied_folders");
+  let main_contents =
+    std::fs::read_to_string(test_folder_path.join("main.ts")).unwrap();
+  temp_dir.write("./main.ts", main_contents);
+
+  let deno = util::deno_cmd_with_deno_dir(&deno_dir)
+    .current_dir(temp_dir.path())
+    .arg("run")
+    .arg("--unstable")
+    .arg("-A")
+    .arg("main.ts")
+    .envs(env_vars())
+    .stdout(Stdio::piped())
+    .stderr(Stdio::piped())
+    .spawn()
+    .unwrap();
+  let output = deno.wait_with_output().unwrap();
+  assert!(output.status.success());
+
+  let expected_initial_output =
+    std::fs::read_to_string(test_folder_path.join("main.out")).unwrap();
+
+  assert_eq!(
+    String::from_utf8(output.stderr).unwrap(),
+    expected_initial_output
+  );
+
+  assert!(temp_dir.path().join("deno.lock").exists());
+  let grandchild_path = deno_dir
+    .path()
+    .join("npm")
+    .join("localhost_4545")
+    .join("npm")
+    .join("registry")
+    .join("@denotest")
+    .join("peer-dep-test-grandchild");
+  assert!(grandchild_path.join("1.0.0").exists());
+  assert!(grandchild_path.join("1.0.0_1").exists()); // copy folder, which is hardlinked
+
+  // run again
+  let deno = util::deno_cmd_with_deno_dir(&deno_dir)
+    .current_dir(temp_dir.path())
+    .arg("run")
+    .arg("--unstable")
+    .arg("-A")
+    .arg("main.ts")
+    .envs(env_vars())
+    .stdout(Stdio::piped())
+    .stderr(Stdio::piped())
+    .spawn()
+    .unwrap();
+  let output = deno.wait_with_output().unwrap();
+  assert_eq!(String::from_utf8(output.stderr).unwrap(), "1\n2\n");
+  assert!(output.status.success());
+
+  // Ensure it works with reloading. This output will be slightly different
+  // because resolution has already occurred due to the lockfile.
+  let expected_reload_output =
+    std::fs::read_to_string(test_folder_path.join("main_reload.out")).unwrap();
```

Comment:
Content of `main_reload.out` and `main.out` seems to be identical, am I missing something here?

Reply:
Oh, it wasn't until I fixed some bugs 😅